pyhms/sprout/sprout_filters.py

```diff
@@ -60,25 +57,26 @@ def __init__(self, min_distance_factor: float = 2.0, norm_ord: int = 2) -> None:
 
     def __call__(
         self,
-        candidates: Dict[AbstractDeme, Tuple[Dict[str, float], List[Individual]]],
+        candidates: dict[AbstractDeme, DemeCandidates],
         tree,
-    ) -> Dict[AbstractDeme, Tuple[Dict[str, float], List[Individual]]]:
-        assert (
-            "NBC_mean_distance" in next(iter(candidates.values()))[0]
+    ) -> dict[AbstractDeme, DemeCandidates]:
+        assert all(
+            [cand.features.NBC_mean_distance is not None for cand in candidates.values()]
         ), "NBC_FarEnough filter requires NBC_mean_distance feature in candidates added throuhg NBC_Generator"
 
+        def nbc_dist(ind, centroid, mean_dist):
```

Comment:
can't we move this function outside of this class? or define it as a classmethod?

Reply:
this function returns a bool, the name should be adjusted to reflect this

Reply:
+ add type hints